solr/core/src/java/org/apache/solr/servlet/CoordinatorHttpSolrCall.java

```diff
@@ -92,8 +93,9 @@ public static SolrCore getCore(
                 "synthetic collection: {} does not exist, creating.. ", syntheticCollectionName);
           }
           createColl(syntheticCollectionName, solrCall.cores, confName);
+          // Try to get core again now that we have created the synthetic core
+          core = solrCall.getCoreByCollection(collectionName, isPreferLeader);
         }
-        SolrCore core = solrCall.getCoreByCollection(syntheticCollectionName, isPreferLeader);
```

Comment:
this line we removed has `syntheticCollectionName`, which you correct to `collectionName`
```
 SolrCore core = solrCall.getCoreByCollection(syntheticCollectionName, isPreferLeader);
```